tests/queries/tests.py

```diff
@@ -1314,7 +1314,7 @@ def test_ticket7107(self):
     def test_ticket8683(self):
         # An error should be raised when QuerySet.datetimes() is passed the
         # wrong type of field.
-        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateTimeField."):
+        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateField or DateTimeField."):
```

Comment:
Good catch. I'll write a test to ensure this cannot happen. If the current API does allow dates into datetimes() then I can just query _meta for the field and do the check in the frontend method.

I haven't checked why you wrote two methods, but I suspect it's because tzinfo is only available on the datetimes() method. I'll double check tomorrow, as I've still got a couple of things to tidy up.